### PR TITLE
Feat/xai explainer

### PR DESCRIPTION
**Pull request by levente-daczo**
### Dependencies:

- Added the react-markdown package to support rendering markdown content.

### New Features:

- Introduced an Explainer page that allows users to input a repository URL and get an explanation. This is backed by an API integration to fetch the explanations.
- Added a new helper API assistantApi which communicates with the explainer API endpoint.
- The form fetches explanations using the EventSource API, providing a stream of data until the explanation is complete.
- Markdown content fetched from the explainer API is rendered using the react-markdown component.

### Configuration:

- Added a new configuration constant **explainerApiUrl** to store the base URL of the explainer API.
- Updated the routes configuration to include the new Explainer page.

### CODE REVIEW

**Comment by CiprianDraghici** [comment]
Please rebase on top of `development` branch

File: src/components/Template/hooks/useNavigation/useNavigation.ts [hunk 1]
```diff
@@ -33,6 +34,10 @@ export const useNavigation = () => {
     {
       path: '/smart-contract',
       icon: faFile
+    },
+    {
+      path: '/explainer',
```
Comment (arhtudormorar) [inline src/components/Template/hooks/useNavigation/useNavigation.ts:39]: use routeNames for paths

File: src/pages/Explainer/components/ExplainForm.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    []
+  );
+
+  const handleCodeExplanationStreamFinished = useCallback(() => {
+    eventSource?.close();
+    setShowLoader(false);
+  }, [eventSource]);
+
+  useEffect(() => {
+    console.log('useEffect', eventSource);
```
Comment (arhtudormorar) [inline src/pages/Explainer/components/ExplainForm.tsx:53]: log

**Review by arhtudormorar — APPROVED** [review]

**Comment by CiprianDraghici** [comment]
Please add a description of this feature. Also, please test it before making the PR ready for review. It looks like something is not working at the backend level. Check this out
Screenshot 2023-10-20 at 13 45 25

File: src/components/Template/hooks/useNavigation/useNavigation.ts (hunk 1 above)
Comment (levente-daczo) [reply src/components/Template/hooks/useNavigation/useNavigation.ts:39]: I followed the current pattern when adding the route. I couldn't locate any **routeNames** in the project. Could you provide an example or clarify your suggestion?

Comment (CiprianDraghici) [reply src/components/Template/hooks/useNavigation/useNavigation.ts:39]: Hey @levente-daczo , please fetch the latest changes from the `development` branch. There is a const in routes.ts file. It was introduced recently.

**Comment by levente-daczo** [comment]
The issue with the backend was resolved. If I understood correctly there was a symlink issue in the deployment script.

File: src/config.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
 export const applicationName = 'Utils';
 export const companyName = 'MultiversX';
 export const companyWebsite = 'https://multiversx.com/';
+export const explainerApiUrl = 'https://devnet-tools.multiversx.com/xbot/api/xai';
```
Comment (CiprianDraghici) [inline src/config.tsx:4]: This should be computed related to the selected network (`mainnet`, `devnet`, `testnet`).
eg. 
for `mainnet` => `https://tools.multiversx.com/xbot/api/xai;`
for `devnet` => `https://devnet-tools.multiversx.com/xbot/api/xai;`
for `testnet` => `https://testnet-tools.multiversx.com/xbot/api/xai;`

There is a hook named useChain() which returns the selected network. You can use this hook in order to compute the explainerApiURL. An idea would be to store the endpoint as a constant in the config.ts file (`export const explainerApiUrl = "/xbot/api/xai"`) and compute only the BASE URL

**Comment by CiprianDraghici** [comment]
> The issue with the backend was resolved. If I understood correctly there was a symlink issue in the deployment script.

Tried but still not working. Maybe is not deployed yet...
I added as an input this URL: https://github.com/multiversx/mx-ping-pong-sc

**Comment by levente-daczo** [comment]
> > The issue with the backend was resolved. If I understood correctly there was a symlink issue in the deployment script.
> 
> Tried but still not working. Maybe is not deployed yet... Used this input: https://github.com/multiversx/mx-ping-pong-sc

The url that should be used for that contract is [https://github.com/multiversx/mx-ping-pong-sc/tree/main/ping-pong](https://github.com/multiversx/mx-ping-pong-sc/tree/main/ping-pong)

Reasoning:
A GitHub repository can contain more than one SC and more than on SC version based on the branch.

An update for the form header and tooltip was made to be more specific.

File: src/pages/Explainer/components/ExplainForm.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [eventSource, setEventSource] = useState<EventSource | null>(null);
+  const { chain } = useChain();
+
+  const initialValues = { repositoryUrl: '' };
+
+  const handleEventSourceError = useCallback(() => {
+    eventSource?.close();
+    setShowLoader(false);
+    setErrorMessage('Explainer encounterd an unknown error. Please try again.');
```
Comment (CiprianDraghici) [inline src/pages/Explainer/components/ExplainForm.tsx:27]: typo: encounterd -> encountered

File: src/helpers/assistantApi.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    },
+}
+
+const AssistantApiEndpoints = {
+    utils: {
+        smartContractExplanation: "/utils/smart_contract_explanation/",
+    },
+}
+
+const ApiBaseUrls = {
```
Comment (CiprianDraghici) [inline src/helpers/assistantApi.ts:24]: This constant could be moved into config.ts file

**Review by CiprianDraghici — APPROVED** [review]

**Comment by CiprianDraghici** [comment]
Thanks for your contribution!

**Pull request merged**